File: PHP/2352. Equal Row and Column Pairs.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[][]  $grid
     */
    public function equalPairs(array $grid): int
    {
        $rows = [];
        $cols = [];
        $length = count($grid);

        for ($i = 0; $i < $length; $i++) {
            $str = implode('+', $grid[$i]);
            $rows[$str] = isset($rows[$str]) ? $rows[$str] + 1 : 1;

            for ($j = 0; $j < $length; $j++) {
                $cols[$i][] = $grid[$j][$i];
            }
        }

        $count = 0;
        for ($i = 0; $i < $length; $i++) {
            $str = implode('+', $cols[$i]);

            if (array_key_exists($str, $rows)) {
                $count += $rows[$str];
            }
        }

        return $count;
    }
}

File: Go/2352. Equal Row and Column Pairs.go
package main

//goland:noinspection GoUnusedFunction
func equalPairs(grid [][]int) int {
	length := len(grid)
	rows := make(map[[200]int]int)
	cols := [200]int{}

	for i := 0; i < length; i++ {
		copy(cols[:], grid[i])
		rows[cols]++
	}

	pairs := 0

	for i := 0; i < length; i++ {
		cols := [200]int{}
		for j := 0; j < length; j++ {
			cols[j] = grid[j][i]
		}

		if v, exists := rows[cols]; exists {
			pairs += v
		}
	}

	return pairs
}
